File: tests/Circuit.Interop.Tests/DocumentationExamples/SkillsSecurity.cs
using Circuit;

namespace Circuit.Interop.Tests.DocumentationExamples;

internal static class SkillsSecurityExample
{
    public static ResolvedSkill CreateResolvedSkill()
    {
        var reference = SkillReference.CreateInline(
            "skill.inline-style",
            "1.0.0",
            "Use a calm, concise support tone.",
            "Inline support guidance.");

        return new ResolvedSkill(
            reference,
            new Dictionary<string, object?>
            {
                ["audience"] = "premium-support",
                ["revision"] = 3,
            });
    }

    public static SkillReference CreateFileSkill()
        => SkillReference.CreateFile(
            "skill.file-style",
            "1.0.0",
            ["/srv/circuit/skills/support-style"],
            "File-backed guidance with SKILL.md and optional references.");
}

File: tests/Circuit.FSharp.Tests/DocumentationExamples/SkillsSecurity.fs
namespace Circuit.FSharp.Tests.DocumentationExamples

open System.Collections.Generic
open Circuit.Core

module SkillsSecurityExample =
    let inlineSkill =
        SkillReference.Create(
            "skill.inline-style",
            "1.0.0",
            "Inline support guidance.",
            SkillSource.CreateInline(
                "Use a calm, concise support tone.",
                [ SkillResource.Create("glossary", box "vip = high-touch customer") ],
                [ SkillScriptDescriptor.Create("normalize-contact") ]
            )
        )

    let resolvedSkill =
        ResolvedSkill.Create(
            inlineSkill,
            [ KeyValuePair("audience", box "premium-support")
              KeyValuePair("revision", box 3) ]
        )

    let fileSkill =
        SkillReference.Create(
            "skill.file-style",
            "1.0.0",
            "File-backed guidance with SKILL.md and optional references.",
            SkillSource.CreateFile "/srv/circuit/skills/support-style"
        )
